suite-native/module-send-receive/src/screens/ReceiveModalScreen.tsx

```diff
@@ -0,0 +1,80 @@
+import React, { ReactNode, useCallback, useEffect, useMemo, useState } from 'react';
+
+import { S } from '@mobily/ts-belt';
+
+import {
+    RootStackParamList,
+    RootStackRoutes,
+    Screen,
+    ScreenHeader,
+    StackProps,
+} from '@suite-native/navigation';
+
+import { AccountSelectionStep } from '../components/AccountSelectionStep';
+import { sendReceiveContentType, SendReceiveContentType } from '../contentType';
+import { AddressGenerationStep } from '../components/AddressGenerationStep';
+import { AddressConfirmationStep } from '../components/AddressConfirmationStep';
+import { FreshAddressStep } from '../components/FreshAddressStep';
+
+const DEFAULT_CONTENT_TYPE = sendReceiveContentType.selectAccountToReceive;
+
+export const ReceiveModalScreen = ({
+    route,
+    navigation,
+}: StackProps<RootStackParamList, RootStackRoutes.ReceiveModal>) => {
+    const [selectedAccountKey, setSelectedAccountKey] = useState<string>(
+        route.params.accountKey || '',
+    );
+    const [contentType, setContentType] = useState<SendReceiveContentType>(DEFAULT_CONTENT_TYPE);
+
+    useEffect(() => {
+        if (S.isNotEmpty(selectedAccountKey)) {
+            setContentType(sendReceiveContentType.createNewAddressToReceive);
+        }
+    }, [selectedAccountKey]);
+
+    const handleChangeContentType = (type: SendReceiveContentType) => {
```

Comment:
this handler seems redundant why not use directly `setContentType` because their signature is same